### PR TITLE
The initial implementation of the calculator UI

### PR DESCRIPTION
**Pull request by wixette**
Only implemented some fundamental interactions:

* The basic display feature set of the calculator screen.
* The handler of the calculator buttons.
* The sound effect of button clicks.
* The cached input buffer of the calculator - without validating or executing input expressions.

### CODE REVIEW

**Comment by codingpotato** [comment]
One general comment: 
At first I cannot run this project in Unity3D IDE. The reason is I didn't install Git LFS, all the files stored with Git LFS are not download correctly. I have to install Git LFS according to the instruction: https://docs.github.com/en/github/managing-large-files/versioning-large-files/installing-git-large-file-storage

Suggest to add an instruction in README.md for how to use Git LFS.

File: Assets/Scripts/Calculator.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+// The calculator that executes input expressions.
+public class Calculator {
+  private const int _maxChars = 100;
+  private readonly CalculatorScreen _screen = null;
+  private readonly StringBuilder _cachedInput = new StringBuilder();
+
+  private bool _hasError = false;
+
+  public bool HasError => _hasError;
```
Comment (codingpotato) [inline Assets/Scripts/Calculator.cs:25]: If "public bool HasError { get; private set; } = false" simpler than these two lines of code?

File: Assets/Scripts/CalculatorInput.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  public const string Del = "Del";
+  public const string Add = "+";
+  public const string Sub = "-";
+  public const string Mul = "*";
+  public const string Div = "/";
+  public const string Dot = ".";
+  public const string Equal = "=";
+
+  // A map from input commands to the corresponding printable strings. Inputs like "AC" or "Del"
+  // have no printable strings.
```
Comment (codingpotato) [inline Assets/Scripts/CalculatorInput.cs:31]: I guess Equal also have no printable strings. Shall we add it in the comment as well?

File: Assets/Scripts/Calculator.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      return;
+    }
+
+    if (CalculatorInput.TryGetPrintable(input, out string printable)) {
+      _cachedInput.Append(printable);
+      if (_cachedInput.Length > _maxChars) {
+        ReportError(CalculatorScreen.Error.Overflow);
+      } else {
+        // TODO: invoke the underlying SeedLang engine to validate the cached expression.
+        _screen.Print(_cachedInput.ToString());
```
Comment (codingpotato) [inline Assets/Scripts/Calculator.cs:48]: Just thinking about the calculator model needs notify the screen to refresh in several places. Is there any kind of MVVM framework in Unity3D to bind views and models to make the notification simpler. Do you think if we need this kind of framework when the UI is more complex?

**Review by codingpotato — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Calculator.cs (hunk 1 above)
Comment (wixette) [reply Assets/Scripts/Calculator.cs]: Done.

File: Assets/Scripts/CalculatorInput.cs (hunk 2 above)
Comment (wixette) [reply Assets/Scripts/CalculatorInput.cs]: Done.

File: Assets/Scripts/Calculator.cs (hunk 3 above)
Comment (wixette) [reply Assets/Scripts/Calculator.cs:46]: Yes, I agree that an MVVM/MVC pattern will be required once the logic becomes more complex.

There are a few 3rd-party MVC libraries but it seems none of them is popular and actively maintained enough. A typical way to implement MVVM in Unity is to leverage the event-dispatcher-handler pattern, e.g., to use System.EventHandler delegate to connect the model class and the corresponding game objects.

The calculator app has a rather simple logic as for the model/view relationship. I will keep this in mind if the relationship turns to be complex in the future.

**Comment by wixette** [comment]
Also added the info about Git LFS in README.md. Please take another look. Thanks,

**Review by codingpotato — APPROVED** [review]

**Pull request merged**